From 52786fe93922e8451f29650059ff8d41322d1d6c Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyunk@xilinx.com>
Date: Thu, 3 Oct 2013 18:49:31 -0700
Subject: [PATCH 398/827] drm: xilinx: osd: Remove xilinx_osd_remove()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Remove xilinx_osd_remove() as devm_* managed functions are used
in xilinx_osd.c.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
(cherry picked from commit 78a58dbe95b02e2bcaebfdf4256e6d9ce20c20fb)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_plane.c | 5 -----
 drivers/gpu/drm/xilinx/xilinx_osd.c       | 5 -----
 drivers/gpu/drm/xilinx/xilinx_osd.h       | 1 -
 3 files changed, 11 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
index e1c598f..e8646e7 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
@@ -545,9 +545,4 @@ void xilinx_drm_plane_remove_manager(struct xilinx_drm_plane_manager *manager)
 			manager->planes[i] = NULL;
 		}
 	}
-
-	if (manager->osd) {
-		xilinx_osd_disable(manager->osd);
-		xilinx_osd_remove(manager->osd);
-	}
 }
diff --git a/drivers/gpu/drm/xilinx/xilinx_osd.c b/drivers/gpu/drm/xilinx/xilinx_osd.c
index 66b096a..b8862a7 100644
--- a/drivers/gpu/drm/xilinx/xilinx_osd.c
+++ b/drivers/gpu/drm/xilinx/xilinx_osd.c
@@ -356,8 +356,3 @@ struct xilinx_osd *xilinx_osd_probe(struct device *dev,
 
 	return osd;
 }
-
-void xilinx_osd_remove(struct xilinx_osd *osd)
-{
-	xilinx_osd_reset(osd);
-}
diff --git a/drivers/gpu/drm/xilinx/xilinx_osd.h b/drivers/gpu/drm/xilinx/xilinx_osd.h
index 8da4400..da29988 100644
--- a/drivers/gpu/drm/xilinx/xilinx_osd.h
+++ b/drivers/gpu/drm/xilinx/xilinx_osd.h
@@ -60,6 +60,5 @@ struct device_node;
 
 struct xilinx_osd *xilinx_osd_probe(struct device *dev,
 				    struct device_node *node);
-void xilinx_osd_remove(struct xilinx_osd *osd);
 
 #endif /* _XILINX_OSD_H_ */
-- 
2.9.3

